src/core.js

```diff
@@ -432,17 +430,20 @@ function(i, name) {
 });
 
 function isArraylike( obj ) {
+	var type, length;
+
+	if ( obj == null ||
+		( type = jQuery.type( obj ) ) === "function" ||
+		type === "number" ||
+		jQuery.isWindow( obj ) ) {
+		return false;
+	}
 
 	// Support: iOS 8.2 (not reproducible in simulator)
 	// `in` check used to prevent JIT error (gh-2145)
 	// hasOwn isn't used here due to false negatives
 	// regarding Nodelist length in IE
-	var length = "length" in obj && obj.length,
-		type = jQuery.type( obj );
-
-	if ( type === "function" || jQuery.isWindow( obj ) ) {
-		return false;
-	}
+	length = "length" in obj && obj.length;
```

Comment:
Those failures are acceptable, because strings and booleans would never end up here from valid function input.